CSV/XML import matching option, refs #10054

sbreker opened the pull request:
CSV Import:

Modified --update option to be able to take parameters to modify the update
logic. Adding further parameters for --update can leverage work done for this
task. Parameters will be specified as: --update="match". Parameters are
verified and an exception will be thrown if an unknown parameter is specified.

Added logic to implement 'match' parameter with --update option. When set,
Information Objects must match to a record in the database or they will be
skipped and a message will be logged in the console.

sbreker committed:
CSV/XML import matching option, refs #10054

CSV Import:

Modified --update option to be able to take parameters to modify the update
logic. Adding further parameters for --update can leverage work done for this
task. Parameters will be specified as: --update="match". Parameters are
verified and an exception will be thrown if an unknown parameter is specified.

Added logic to implement 'match' parameter with --update option. When set,
Information Objects must match to a record in the database or they will be
skipped and a message will be logged in the console.

MikeFE commented on lib/QubitFlatfileImport.class.php:
```diff
@@ -658,8 +659,24 @@ public function row($row = array())
+          }
+          else
+          {
+            // create new object
```
hehe I realize this is a legacy comment, but it's silly. I'd remove it.

MikeFE commented on lib/task/import/csvImportTask.class.php:
```diff
@@ -1074,6 +1074,27 @@ public function execute($arguments = array(), $options = array())
+      switch ($options['update'])
+      {
+        case 'match':
+          // Save match option. If update is ON, and match is set, only updating
```
good comment

MikeFE commented on lib/task/import/csvImportBaseTask.class.php:
```diff
@@ -77,6 +77,21 @@ protected function validateOptions($options)
+      // if supplied param is not in $validParams, throw exception
+      if (!in_array(trim($options['update']), $validParams))
+      {
+        $msg = sprintf("Parameter \"%s\" is not valid for --update option.", $options['update']);
```
I'd use single quotes here instead of double, so you don't need to do the backslash escaping.

sbreker committed:
CSV/XML import matching option, refs #10054

XML/EAD Import:

Added new --match option. When set, Information Objects must match to a
record in the database or they will be skipped and a message will be logged
in the console.

sbreker committed:
CSV/XML import matching option, refs #10054

Removed one comment in QubitFlatfileImport.class.php.

Changed double quotes to single in csvImportBaseTask.class.php.

MikeFE commented on lib/QubitXmlImport.class.php:
```diff
@@ -414,7 +418,7 @@ private function populateObject(&$domNode, &$importDOM, &$mapping, &$currentObje
     $doSave = true;
     // if this is an information object in an XML EAD import, run the enhanced matching check.
-    if ($currentObject instanceof QubitInformationObject && $importSchema == 'ead')
+    if ($currentObject instanceof QubitInformationObject && $importSchema == 'ead' &&  $this->options['match'])
```
If the key 'match' isn't in $this->options I think it will throw a PHP notice or warning (sometimes can't see them without error_reporting(E_ALL) :/). I recommend using `isset($this->options['match']) && $this->options['match']`

MikeFE commented:
:+1:

sbreker committed:
CSV/XML import matching option, refs #10054

Completed change to protect reference to $this->options['match'] with
isset().
